src/Analyzers/CSharp/CodeFixes/UseCollectionInitializer/CSharpUseCollectionInitializerCodeFixProvider.cs

```diff
@@ -36,80 +42,198 @@ public CSharpUseCollectionInitializerCodeFixProvider()
         }
 
         protected override StatementSyntax GetNewStatement(
+            SourceText sourceText,
             StatementSyntax statement,
             BaseObjectCreationExpressionSyntax objectCreation,
-            ImmutableArray<ExpressionStatementSyntax> matches)
+            int wrappingLength,
+            bool useCollectionExpression,
+            ImmutableArray<Match<StatementSyntax>> matches)
         {
             return statement.ReplaceNode(
                 objectCreation,
-                GetNewObjectCreation(objectCreation, matches));
+                GetNewObjectCreation(sourceText, objectCreation, wrappingLength, useCollectionExpression, matches));
         }
 
-        private static BaseObjectCreationExpressionSyntax GetNewObjectCreation(
+        private static ExpressionSyntax GetNewObjectCreation(
+            SourceText sourceText,
             BaseObjectCreationExpressionSyntax objectCreation,
-            ImmutableArray<ExpressionStatementSyntax> matches)
+            int wrappingLength,
+            bool useCollectionExpression,
+            ImmutableArray<Match<StatementSyntax>> matches)
         {
-            return UseInitializerHelpers.GetNewObjectCreation(
-                objectCreation, CreateExpressions(objectCreation, matches));
+            return useCollectionExpression
+                ? CreateCollectionExpression(objectCreation, matches, MakeMultiLine(sourceText, objectCreation, matches, wrappingLength))
+                : CreateObjectInitializerExpression(objectCreation, matches);
         }
 
-        private static SeparatedSyntaxList<ExpressionSyntax> CreateExpressions(
+        private static BaseObjectCreationExpressionSyntax CreateObjectInitializerExpression(
             BaseObjectCreationExpressionSyntax objectCreation,
-            ImmutableArray<ExpressionStatementSyntax> matches)
+            ImmutableArray<Match<StatementSyntax>> matches)
         {
-            using var _ = ArrayBuilder<SyntaxNodeOrToken>.GetInstance(out var nodesAndTokens);
+            var expressions = CreateElements(objectCreation, matches, static (_, e) => e);
+            var withLineBreaks = AddLineBreaks(expressions, includeFinalLineBreak: true);
+            return UseInitializerHelpers.GetNewObjectCreation(objectCreation, withLineBreaks);
+        }
 
-            UseInitializerHelpers.AddExistingItems(objectCreation, nodesAndTokens);
+        private static CollectionExpressionSyntax CreateCollectionExpression(
+            BaseObjectCreationExpressionSyntax objectCreation,
+            ImmutableArray<Match<StatementSyntax>> matches,
+            bool makeMultiLine)
+        {
+            var elements = CreateElements<CollectionElementSyntax>(
+                objectCreation, matches,
+                static (match, expression) => match?.UseSpread is true ? SpreadElement(expression) : ExpressionElement(expression));
```

Comment:
similar behavior to normal collection initializers.  however, here we may produce spread elements, not just normal expressions.  the 'Match' value tells us if we want that.  'match' is nullable as some of the expressions provided here came from the *original* object creation expression, and thus have no match data.